### ISSUE TITLE
Problems with fullscreen window using DirectX11 gamma functionality

### ISSUE THREAD

**Issue by SDLBugzilla** [libsdl-org/SDL#2045]
# This bug report was migrated from our old Bugzilla tracker.

**Reported in version:** 2.0.3
**Reported for operating system, platform:** Windows 7, x86_64

# Comments on the original bug report:

On 2015-12-07 16:04:55 +0000,  wrote:

> We are using SDL with DirectX11 and the gamma functionality of DirectX. 
> 
> When switching to fullscreen mode and changing the gamma everything works fine.
> 
> But when pressing alt + tab and then restoring the window, the window does not switch to fullscreen mode properly, the gamma doesn't work anymore.
> 
> DirectX outputs the error: DXGI ERROR: IDXGIOutput::GetGammaControlCapabilities: This function can only be called if TakeOwnership has been called. [ MISCELLANEOUS ERROR # 47: ]
> 
> But calling TakeOwnership doesn't solve the problem. Switching back to window mode and then back to fullscreen solves the problem.

**Comment by obfuscate** [comment]
Hello!

It seems I have the same problem. We have the next case:
1. Start the application in the fullscreen mode on a display 1.
2. Move the application from the display 1 to a display 2 (using shift+win+arrow).
3. Alt-tab to switch to another application. Alt-tab to switch back to our application.
Result: The gamma is changed.

I also noticed a strange behavior: restoring window always has the next sequence of events: maximized, restore, exposed. I react on the last event and call setGammRamp, but it sometimes works, sometimes doesn't.

**Comment by obfuscate** [comment]
I'm not sure that it is the same problem, but I also noticed the next behavior:
I called setGammaRamp reacting of WINDOWS_ENTER\LEAVE events (1.0f/2.2f for enter, 2.2f for leave, for example).
1. Start the application in the fullscreen mode on a display 1.
2. Move the application from the display 1 to a display 2 (using shift+win+arrow).
3. Move the cursor in the application and out.
Result: You'll notice that gamma affects the display 1, not 2.
